pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -483,6 +483,70 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         applyNCMultiQubitOp(arr, num_qubits, matrix, {}, {}, wires, inverse);
     }
 
+    /**
+     * @brief Apply a sparse matrix with controls to the statevector.
+     * @tparam PrecisionT Floating point precision of underlying statevector
+     * data
+     * @tparam index_type Index type
+     * @param arr Pointer to the statevector.
+     * @param num_qubits Number of qubits.
+     * @param row_map_ptr Pointer to the row map.
+     * @param col_idx_ptr Pointer to the columns indexes
+     * @param values_ptr Pointer to the values.
+     * @param controlled_wires Control wires.
+     * @param controlled_values Control values (false or true).
+     * @param wires Wires the gate applies to.
+     * @param inverse Indicate whether inverse should be taken.
+     */
+    template <class PrecisionT, class index_type = std::size_t>
```

Comment:
Just a fly-by comment, the template parameters should follow CamelCase to make it easier to distinguish them from variables (https://github.com/PennyLaneAI/guidance-docs/blob/master/development/cpp-dev-guidelines.md#naming-conventions). Maybe `IndexT`, `IdxT` or similar could work here?

Reply:
My bad, I'm doing it in `IndexT` in other parts. I will fix that.

Reply:
While I'm working on that, I will see if I can update index template parameters everywhere. I think this was the reason for my "bamboozlement".

Reply:
Hey @mlxd, I updated this template parameter name everywhere I found. CIs are looking good so far! 
Thank you for bringing this to my attention.